theano/gof/type.py

```diff
@@ -854,27 +854,44 @@ class EnumType(Type, dict):
 
     """
 
-    def check_ctype(self):
+    def init_ctype(self, ctype):
         # C type may be a list of keywords, e.g. "unsigned long long".
         # We should check each part.
-        if not all(re.match('^[A-Za-z_][A-Za-z0-9_]*$', el) for el in self.ctype.split()):
-            raise TypeError('%s: invalid C type' % type(self).__name__)
+        ctype_parts = ctype.split()
+        if not all(re.match('^[A-Za-z_][A-Za-z0-9_]*$', el) for el in ctype_parts):
+            raise TypeError('%s: invalid C type.' % type(self).__name__)
+        self.ctype = ' '.join(ctype_parts)
 
     def __init__(self, **kwargs):
-        self.ctype = kwargs.pop('ctype', 'double')
-        self.check_ctype()
+        self.init_ctype(kwargs.pop('ctype', 'double'))
+        self.aliases = dict()
         for k in kwargs:
             if re.match('^[A-Z][A-Z0-9_]*$', k) is None:
                 raise AttributeError('%s: invalid enum name: "%s". '
                                      'Only capital letters, underscores and digits '
                                      'are allowed.' % (type(self).__name__, k))
+            if isinstance(kwargs[k], (list, tuple)):
+                if len(kwargs[k]) != 2:
+                    raise TypeError('%s: when using a tuple to define a constant, your tuple should contain 2 values: '
+                                    'constant alias followed by constant value.' % type(self).__name__)
+                nickname, value = kwargs[k]
+                if not isinstance(nickname, str):
+                    raise TypeError('%s: constant alias should be a string, got "%s".'
+                                    % (type(self).__name__, nickname))
+                if nickname in self.aliases:
+                    raise TypeError('%s: consant alias "%s" already used.' % (type(self).__name__, nickname))
+                self.aliases[nickname] = k
+                kwargs[k] = value
             if isinstance(kwargs[k], bool):
```

Comment:
In fact, I want to pass through this IF in any case: if kwargs[k] is a tuple above, the constant alias is extracted, and then the real constant value is put to kwargs[k], then it should be checked again here.